hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/erasurecode/rawcoder/DecodingState.java

```diff
@@ -39,7 +39,7 @@ <T> void checkParameters(T[] inputs, int[] erasedIndexes,
                            T[] outputs) {
     if (inputs.length != decoder.getNumParityUnits() +
         decoder.getNumDataUnits()) {
-      throw new IllegalArgumentException("Invalid inputs length");
+      throw new HadoopIllegalArgumentException("Invalid inputs length");
```

Comment:
implicit changes in public api; revert

Reply:
Thanks @steveloughran for the review. I have reverted the change.